src/na/device/CMakeLists.txt

```
@@ -0,0 +1,148 @@
+
+  # Make version available
+  target_compile_definitions(${TARGET_NAME} PRIVATE MQT_CORE_VERSION="${MQT_CORE_VERSION}")
+endif()
+
+# Set target name
+set(TARGET_NAME mqt-core-qdmi-na-device-generator)
+
+# If the target is not already defined
+if(NOT TARGET ${TARGET_NAME})
+  # Add executable for device generation
+  add_executable(${TARGET_NAME})
+
+  # add sources to target
+  target_sources(${TARGET_NAME} PRIVATE App.cpp)
+
+  # Link Generator library
+  target_link_libraries(${TARGET_NAME} PRIVATE MQT::CoreNaDeviceGen spdlog::spdlog)
+
+  # set versioning information
+  set_target_properties(${TARGET_NAME} PROPERTIES VERSION ${PROJECT_VERSION} EXPORT_NAME
+                                                                             CoreNaDeviceGen)
+  # place in the bin directory
+  set_target_properties(${TARGET_NAME} PROPERTIES RUNTIME_OUTPUT_DIRECTORY
+                                                  "${CMAKE_BINARY_DIR}/bin")
+
+  # set c++ standard
+  target_compile_features(${TARGET_NAME} PRIVATE cxx_std_17)
+
+  # Make version available
+  target_compile_definitions(${TARGET_NAME} PRIVATE MQT_CORE_VERSION="${MQT_CORE_VERSION}")
+
+  # Create an alias for the target
+  add_executable(MQT::CoreQdmiNaDeviceGenerator ALIAS ${TARGET_NAME})
+endif()
+
+# Set target name
+set(TARGET_NAME ${MQT_CORE_TARGET_NAME}-na-device)
+
+# Set prefix for QDMI
+set(QDMI_PREFIX "MQT_NA")
+
+# If the target is not already defined
+if(NOT TARGET ${TARGET_NAME})
+
+  # Set paths
+  set(JSON_FILE ${PROJECT_SOURCE_DIR}/json/na/device.json)
+  set(DEVICE_HDR ${CMAKE_CURRENT_BINARY_DIR}/include/Initializer.hpp)
+
+  # Generate definitions for device
+  add_custom_command(
+    OUTPUT ${DEVICE_HDR}
+    COMMAND MQT::CoreQdmiNaDeviceGenerator ARGS generate --output ${DEVICE_HDR} ${JSON_FILE}
+    DEPENDS ${JSON_FILE}
+    COMMENT "Generating C++ header from ${JSON_FILE}")
+  add_custom_target(generate_na_device_header DEPENDS ${DEVICE_HDR})
+
+  # Add library
+  add_mqt_core_library(${TARGET_NAME} ALIAS_NAME NaQDMI)
+  add_dependencies(${TARGET_NAME} generate_na_device_header)
+
+  # Generate prefixed QDMI headers
+  generate_prefixed_qdmi_headers(${QDMI_PREFIX})
+  file(GLOB_RECURSE QDMI_HDRS ${CMAKE_CURRENT_BINARY_DIR}/include/mqt_na_qdmi/**.hpp)
+
+  # add sources to target
+  target_sources(${TARGET_NAME} PRIVATE Device.cpp)
+
+  # add headers using file sets
+  target_sources(
+    ${TARGET_NAME}
+    PUBLIC FILE_SET
+           HEADERS
+           BASE_DIRS
+           ${MQT_CORE_INCLUDE_BUILD_DIR}
+           ${CMAKE_CURRENT_BINARY_DIR}/include
+           FILES
+           ${DEVICE_HDR}
+           ${MQT_CORE_INCLUDE_BUILD_DIR}/na/device/Device.hpp
+           ${QDMI_HDRS})
+
+  # add link libraries
+  target_link_libraries(
+    ${TARGET_NAME}
+    PUBLIC qdmi::qdmi
+    PRIVATE MQT::ProjectOptions MQT::ProjectWarnings spdlog::spdlog)
+
+  # set c++ standard
+  target_compile_features(${TARGET_NAME} PRIVATE cxx_std_17)
```

```suggestion
  target_compile_features(${TARGET_NAME} PRIVATE cxx_std_20)
```
This should really be moved into the `add_mqt_library` macro.